src/sql/workbench/api/node/mainThreadNotebookDocumentsAndEditors.ts

```diff
@@ -592,4 +608,15 @@ export class MainThreadNotebookDocumentsAndEditors extends Disposable implements
 		}
 		return notebookCells;
 	}
+
+	private async doChangeKernel(editor: MainThreadNotebookEditor, displayName: string): Promise<boolean> {
+		let listeners = this._modelToDisposeMap.get(editor.id);
+		editor.model.changeKernel(displayName);
+		return new Promise((resolve) => {
+			listeners.push(editor.model.kernelChanged((kernel) => {
```

Comment:
is there some case where the changeKernel could have already completed & notified before we get to here? Should we set up the listener and promise handler before calling the method?